libtenzir/builtins/components/metrics_collector.cpp

```diff
@@ -139,9 +139,6 @@ class plugin final : public virtual component_plugin {
 
   auto make_component(node_actor::stateful_pointer<node_state> node) const
     -> component_plugin_actor override {
-    if (collect(plugins::get<metrics_plugin>()).empty()) {
-      return {};
-    }
```

Comment:
The assertion in L66 is going to fire when you run into this, so that would now need to be handled gracefully.